### DISCUSSION TITLE
How to use TradingView Stoch (14, 1, 3) with GetStoch of this library

### DISCUSSION THREAD

**Discussion by SWSSolutions** [DaveSkender/Stock.Indicators#1012] [Help and support]
Hi,

I use several indicators from this library and all work fine except one, Stoch. TradingView use the defaults K = 14, D = 1, Smooth = 3 and for some reason I get different results. It looks likes GetStoch(14, 3, 1) comes closest but most of them during the time period are wrong. Any suggestions?

What have I tried:
- Different parameters of GetStoch;
- Short (12 hours with a 1 minute time-frame), middle (1 day with a 1 minute time-frame) and very large historical quotes (1 year with 1 minute time-frame);

How to reproduce
- Getting data from api.binance.com (https://api.binance.com/api/v3/klines?symbol=BNBBUSD&interval=1m&limit=1000&startTime=1674801170223&endTime=1674844370223);
- Convert them to Quote
- Use GetStoch(14, 3, 1) (or any other setting like GetStoch(14, 1, 3), GetStoch(14, 3, 3) etc)

- Go to: https://www.binance.com/en/trade/BNB_BUSD?theme=dark&type=spot (or https://www.tradingview.com/chart/DVyPeqdv/?symbol=BINANCE%3ABNBBUSD)
- Set the time-frame on 1 minute and set the time to UTC;

- Compare the results

When using the link, the last candle (open on 1674844320000 - Fri Jan 27 2023 18:32:00 GMT+0000) must have a stoch of: k:60 d:80) according to TradingView.
With GetStoch(14, 3, 3) it has a toch of: k:83.33 d:89.44
With GetStoch(14, 1, 3) it has a toch of: k:83.33 d:83.33
With GetStoch(14, 3, 1) it has a toch of: k:70 d:83.33

So i suppose i'm doing something wrong, any directions?

**Comment by DaveSkender** [comment]
Here’s a few things to try:
- check out: [common issues people have when comparing to TradingView](https://github.com/DaveSkender/Stock.Indicators/discussions/801)
- confirm that you are using the parameters in the right sequence, we may not put them in the same order.
- make sure you're using enough quote history.  There are normal convergence related errors if you do not use enough.
- compare with other charting providers.  It’s possible TradingView has an error

**Reply by gabrielblaj** [reply]
Please tell me how you convert data from binance to quote, thank you

**Reply by SWSSolutions** [reply]
I am using the System.Text.Json namespace with a Candle class. After deserialization I convert it to a list of Quotes.

Anyway, stoch is working correctly now, apparently there are different implications (even within TradingView!).

I only have some problems with PSAR, it's not always correct (I would say 5 out of 100 candles) compared to TradingView (no matter what implication). I compared the results to other .Net libraries and all results were the same, so TradingView does something different

**Reply by DaveSkender** [reply]
I isolated a couple clear Parabolic SAR errors in TradingView in #1031 if you want to read more details.  It’s come up a few times, so I needed to explain.
